File: UKOM/Moview/app/src/main/java/com/komputerkit/moview/data/model/Diary.kt
package com.komputerkit.moview.data.model

/**
 * Diary model for user film activity screen
 */
data class Diary(
    val diary_id: Int,
    val movie_id: Int,
    val title: String,
    val year: Int,
    val poster_path: String?,
    val watched_at: String,
    val note: String?,
    val rating: Int,
    val is_liked: Boolean,
    val is_rewatched: Boolean,
    val review_id: Int,
    val review_text: String?,
    val type: String  // "review" or "log"
)

File: UKOM/Moview_backend/app/Models/Diary.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Diary extends Model
{
    protected $table = 'diaries';

    protected $fillable = [
        'user_id',
        'film_id',
        'review_id',
        'watched_at',
        'rating',
        'is_liked',
        'note',
        'is_rewatched',
    ];

    protected $casts = [
        'is_liked'     => 'boolean',
        'is_rewatched' => 'boolean',
        'watched_at'   => 'date',
        'rating'       => 'integer',
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function movie()
    {
        return $this->belongsTo(Movie::class, 'film_id');
    }

    public function review()
    {
        return $this->belongsTo(Review::class, 'review_id');
    }

    public function changeMedias()
    {
        return $this->hasMany(UserChangeMedia::class, 'diaries_id');
    }
}
